resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala

```diff
@@ -254,9 +254,11 @@ private[spark] class ApplicationMaster(args: ApplicationMasterArguments) extends
       var attemptID: Option[String] = None
 
       if (isClusterMode) {
-        // Set the web ui port to be ephemeral for yarn so we don't conflict with
-        // other spark processes running on the same box
-        System.setProperty("spark.ui.port", "0")
+        // Set the web ui port to be ephemeral for yarn if not set explicitly
+        // so we don't conflict with other spark processes running on the same box
+        if (System.getProperty("spark.ui.port") != null) {
```

Comment:
@rajatahujaatinmobi , the check must use '==' operator and not '!='
say spark.ui.port is set to 9999, above if condition equates to true and value is overridden to 0(random port)